deployment/ccip/changeset/v1_6/cs_add_bidirectional_lanes.go

```diff
@@ -0,0 +1,235 @@
+
+			// Setting the fee quoter destination on the source chain
+			if feeQuoterDestUpdatesByChain[laneDef.Source.Selector] == nil {
+				feeQuoterDestUpdatesByChain[laneDef.Source.Selector] = make(map[uint64]fee_quoter.FeeQuoterDestChainConfig)
+			}
+			feeQuoterDestUpdatesByChain[laneDef.Source.Selector][laneDef.Dest.Selector] = laneDef.Dest.FeeQuoterDestChainConfig
+
+			// Setting the destination gas prices on the source chain
+			feeQuoterPriceUpdatesOnSource := feeQuoterPriceUpdatesByChain[laneDef.Source.Selector]
+			if feeQuoterPriceUpdatesOnSource.GasPrices == nil {
+				feeQuoterPriceUpdatesOnSource.GasPrices = make(map[uint64]*big.Int)
+			}
+			feeQuoterPriceUpdatesOnSource.GasPrices[laneDef.Dest.Selector] = laneDef.Dest.GasPrice
+			feeQuoterPriceUpdatesByChain[laneDef.Source.Selector] = feeQuoterPriceUpdatesOnSource
+		}
+	}
+
+	routerMCMSConfig := c.MCMSConfig
+	if c.TestRouter {
+		routerMCMSConfig = nil // Test router is never owned by MCMS
+	}
+
+	return addBidirectionalLanesChangesetConfigs{
+		UpdateFeeQuoterDestsConfig: UpdateFeeQuoterDestsConfig{
+			MCMS:           c.MCMSConfig,
+			UpdatesByChain: feeQuoterDestUpdatesByChain,
+		},
+		UpdateFeeQuoterPricesConfig: UpdateFeeQuoterPricesConfig{
+			MCMS:          c.MCMSConfig,
+			PricesByChain: feeQuoterPriceUpdatesByChain,
+		},
+		UpdateOnRampDestsConfig: UpdateOnRampDestsConfig{
+			MCMS:           c.MCMSConfig,
+			UpdatesByChain: onRampUpdatesByChain,
+		},
+		UpdateOffRampSourcesConfig: UpdateOffRampSourcesConfig{
+			MCMS:           c.MCMSConfig,
+			UpdatesByChain: offRampUpdatesByChain,
+		},
+		UpdateRouterRampsConfig: UpdateRouterRampsConfig{
+			TestRouter:     c.TestRouter,
+			MCMS:           routerMCMSConfig,
+			UpdatesByChain: routerUpdatesByChain,
+		},
+	}
+}
+
+func addBidirectionalLanesPrecondition(e deployment.Environment, c AddBidirectionalLanesConfig) error {
+	configs := c.buildConfigs()
+	state, err := changeset.LoadOnchainState(e)
+	if err != nil {
+		return fmt.Errorf("failed to load onchain state: %w", err)
+	}
+
+	err = configs.UpdateFeeQuoterDestsConfig.Validate(e)
+	if err != nil {
+		return fmt.Errorf("failed to validate UpdateFeeQuoterDestsConfig: %w", err)
+	}
+
+	err = configs.UpdateFeeQuoterPricesConfig.Validate(e)
+	if err != nil {
+		return fmt.Errorf("failed to validate UpdateFeeQuoterPricesConfig: %w", err)
+	}
+
+	err = configs.UpdateOnRampDestsConfig.Validate(e)
+	if err != nil {
+		return fmt.Errorf("failed to validate UpdateOnRampDestsConfig: %w", err)
+	}
+
+	err = configs.UpdateOffRampSourcesConfig.Validate(e, state)
+	if err != nil {
+		return fmt.Errorf("failed to validate UpdateOffRampSourcesConfig: %w", err)
+	}
+
+	err = configs.UpdateRouterRampsConfig.Validate(e, state)
+	if err != nil {
+		return fmt.Errorf("failed to validate UpdateRouterRampsConfig: %w", err)
+	}
+
+	return nil
+}
+
+func addBidirectionalLanesLogic(e deployment.Environment, c AddBidirectionalLanesConfig) (deployment.ChangesetOutput, error) {
```

Comment:
nit: log statements after every success operation. Ignore if there are already log statements available in individual changesets

Reply:
https://github.com/smartcontractkit/chainlink/pull/17015/commits/5059146e9086fc4115d6b3a2a3563beb34c4d991